tests/ctst/steps/bucket-policies/utils.ts

```diff
@@ -0,0 +1,500 @@
+    expectedResultOnAllowTest?: string;
+    excludePermissionOnBucketObjects?: boolean,
+    excludePermissionsOnBucket?: boolean,
+    useWildCardBucketName?: boolean,
+    needsSetup?: boolean,
+};
+
+const needObjectLock = [
+    'GetObjectLegalHold',
+    'PutObjectLegalHold',
+    'GetObjectRetention',
+    'PutObjectRetention',
+    'PutObjectVersionLegalHold',
+    'PutObjectVersionRetention',
+    'GetObjectVersionLegalHold',
+    'GetObjectVersionRetention',
+];
+
+const needObject = [
+    'PutObjectLegalHold',
+    'PutObjectRetention',
+    'PutObjectTagging',
+    'PutObjectAcl',
+    'PutObject',
+    'GetObject',
+    'GetObjectAcl',
+    'GetObjectLegalHold',
+    'GetObjectRetention',
+    'GetObjectTagging',
+    'DeleteObject',
+    'DeleteObjectTagging',
+    'DeleteObjects',
+    'HeadObject',
+    'UploadPart',
+    'UploadPartCopy',
+    'DeleteObjectVersion',
+    'DeleteObjectVersionTagging',
+    'GetObjectVersion',
+    'GetObjectVersionAcl',
+    'GetObjectVersionTagging',
+    'PutObjectVersionAcl',
+    'PutObjectVersionTagging',
+    'PutObjectVersionRetention',
+    'PutObjectVersionLegalHold',
+];
+
+const needVersioning = [
+    'PutBucketReplication',
+    'ListObjectVersions',
+    'DeleteObjectVersion',
+    'DeleteObjectVersionTagging',
+    'GetObjectVersion',
+    'GetObjectVersionAcl',
+    'GetObjectVersionTagging',
+    'PutObjectVersionAcl',
+    'PutObjectVersionTagging',
+    'PutObjectVersionRetention',
+    'PutObjectVersionLegalHold',
+];
+
+const actionPermissions: ActionPermissionsType[] = [
+    // Create bucket is not checking any bucket policy
+    // {
+    //     action: 'CreateBucket',
+    //     permissions: ['s3:CreateBucket'],
+    //     expectedResultOnAllowTest: 'BucketAlreadyOwnedByYou',
+    //     excludePermissionOnBucketObjects: true,
+    //     useWildCardBucketName: true,
+    // },
+    {
+        action: 'PutBucketAcl',
+        permissions: ['s3:PutBucketAcl'],
+        expectedResultOnAllowTest: 'MalformedXML',
+        excludePermissionOnBucketObjects: true,
+    },
+    {
+        action: 'PutBucketEncryption',
+        permissions: ['s3:PutEncryptionConfiguration'],
+        expectedResultOnAllowTest: 'MalformedXML',
+        excludePermissionOnBucketObjects: true,
+    },
+    {
+        action: 'PutBucketLifecycleConfiguration',
+        permissions: ['s3:PutLifecycleConfiguration'],
+        excludePermissionOnBucketObjects: true,
+    },
+    {
+        action: 'PutBucketNotificationConfiguration',
+        permissions: ['s3:PutBucketNotification'],
+        excludePermissionOnBucketObjects: true,
+    },
+    {
+        action: 'PutBucketPolicy',
+        permissions: ['s3:PutBucketPolicy'],
+        excludePermissionOnBucketObjects: true,
+    },
+    {
+        action: 'PutBucketReplication',
+        permissions: ['s3:PutReplicationConfiguration'],
+        expectedResultOnAllowTest: 'InvalidRequest: Versioning must be \'Enabled\'',
```

Comment:
same https://github.com/scality/Integration/blob/development/7.10/tests/bucket-policy/utils/s3APIConfigurations.js#L147

Reply:
This one passes without error:
```
2024-02-21T13:35:11.1210222Z {"reqId":2111,"api":"PutBucketReplicationCommand","input":{"Bucket":"zenko-ctstbucketfortestil4vi","ReplicationConfiguration":{"Role":"arn:aws:iam::123456789012:role/s3-replication-role","Rules":[{"Status":"Enabled","Prefix":"","Destination":{"Bucket":"arn:aws:s3:::examplebucket"}}]}},"response":"{\"$metadata\":{\"httpStatusCode\":200,\"requestId\":\"4cfa37823d40884270de\",\"extendedRequestId\":\"4cfa37823d40884270de\",\"attempts\":1,\"totalRetryDelay\":0}}","accessKey":"XRQ8VHBL2DDHX93SBN0H","secretKey":"zBTN=I7YFRZlcK7WpGHv1ZilZJ/8jZuqQolx2Vhj","isSessionToken":true,"endpoint":"http://s3.zenko.local:80"}
```

I can remove the expected result to be consistent

Reply:
See https://github.com/scality/Zenko/pull/2023/commits/653404951d0bfc4b850d41a112541d0aa7e381d3